### PR TITLE
Delete deprecated apis

### PR DESCRIPTION
**Pull request by krousey**
A work in progress for removing v1beta1 and v1beta2 APIs from the code base. #8087

### CODE REVIEW

File: pkg/api/serialization_test.go [hunk 1]
```diff
@@ -103,8 +95,6 @@ func roundTripSame(t *testing.T, item runtime.Object, except ...string) {
 
 func roundTripAll(t *testing.T, item runtime.Object) {
 	seed := rand.Int63()
-	roundTrip(t, v1beta1.Codec, fuzzInternalObject(t, "v1beta1", item, seed))
-	roundTrip(t, v1beta2.Codec, fuzzInternalObject(t, "v1beta2", item, seed))
```
Comment (bgrant0607) [inline pkg/api/serialization_test.go]: We should be roundtripping v1 here, though that could be done in another PR.

File: pkg/kubectl/cmd/util/helpers_test.go [hunk 2]
```diff
@@ -137,7 +100,7 @@ func TestMerge(t *testing.T) {
 			obj: &api.Service{
 				Spec: api.ServiceSpec{},
 			},
-			fragment: `{ "apiVersion": "v1beta1", "port": 0 }`,
+			fragment: `{ "apiVersion": "v1beta3", "port": 0 }`,
```
Comment (bgrant0607) [inline pkg/kubectl/cmd/util/helpers_test.go]: There's no port field there in v1beta3. Firstly, there's a spec. Secondly, ports are an array.

File: pkg/service/endpoints_controller.go [hunk 3]
```diff
@@ -398,32 +390,14 @@ func (e *EndpointController) checkLeftoverEndpoints() {
...
-	return servicePort
-}
-
-// findPort locates the container port for the given manifest and portName.
-// If the targetPort is a non-zero number, use that.  If the targetPort is 0 or
-// not specified, use the first defined port with the same protocol.  If no port
-// is defined, use the service's port.  If the targetPort is an empty string use
-// the first defined port with the same protocol.  If no port is defined, use
-// the service's port.  If the targetPort is a non-empty string, look that
+// findPort locates the container port for the given pod and portName.  If the
```
Comment (bgrant0607) [inline pkg/service/endpoints_controller.go]: @thockin Is this the expected behavior in v1beta3 and v1?

**Comment by bgrant0607** [comment]
What else needs to be done in this PR?

Please rebase at some point so we can get Travis and Shippable to run the tests. You're getting a merge conflict right now.

File: pkg/kubectl/cmd/util/helpers_test.go (hunk 2 above)
Comment (krousey) [inline pkg/kubectl/cmd/util/helpers_test.go]: Fixed.

**Comment by krousey** [comment]
I rebased and had to modify the code gen tools.

**Comment by krousey** [comment]
Note, I expect on the the travis tests to fails because I haven't updated it to not use v1beta1.

**Comment by bgrant0607** [comment]
@krousey Please replace v1beta1 tests with v1 tests on Travis/Shippable in this PR, assuming that works. If not, just make both builds run v1beta3 tests for now.

**Comment by bgrant0607** [comment]
cc @smarterclayton, so he can verify we're not nuking something Openshift needs.

File: pkg/api/serialization_test.go [hunk 4]
```diff
@@ -137,10 +127,7 @@ func TestList(t *testing.T) {
 
 var nonRoundTrippableTypes = util.NewStringSet("ContainerManifest", "ContainerManifestList")
 var nonInternalRoundTrippableTypes = util.NewStringSet("List", "ListOptions", "PodExecOptions")
-var nonRoundTrippableTypesByVersion = map[string][]string{
-	"PodTemplate":     {"v1beta1", "v1beta2"},
-	"PodTemplateList": {"v1beta1", "v1beta2"},
-}
+var nonRoundTrippableTypesByVersion = map[string][]string{}
```
Comment (bgrant0607) [inline pkg/api/serialization_test.go]: I suggest leaving a comment about the expected format.

File: pkg/service/endpoints_controller.go (hunk 3 above)
Comment (bgrant0607) [inline pkg/service/endpoints_controller.go]: Based on in-person discussion, this seems ok. 

TargetPort is initialized here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/v1beta3/defaults.go#L91

**Comment by bgrant0607** [comment]
LGTM once tests pass.

**Comment by bgrant0607** [comment]
Remove WIP from the PR title once tests are working.

**Comment by smarterclayton** [comment]
@derekwaynecarr can you double check that we're not dependent on any of these types still (pretty sure, just want to verify).

**Comment by ixdy** [comment]
@k8s-bot test this please

(I accidentally wiped the output directory while it was building which caused it to fail.)

**Comment by krousey** [comment]
@bgrant0607 I'll also wait until I get e2e running locally. Need to wait on a heapster release that @vishh is working on.

**Comment by bgrant0607** [comment]
Awesome. LGTM once e2e is working.

File: pkg/service/endpoints_controller.go (hunk 3 above)
Comment (thockin) [inline pkg/service/endpoints_controller.go]: LGTM

Comment (thockin) [inline pkg/service/endpoints_controller.go]: NB: we still have a conflict here if the same port name is used on two containers, but thta is not this PR's problem.

**Comment by thockin** [comment]
This PR is a breath of fresh air.

**Comment by bgrant0607** [comment]
Yes, I am SO dying to nuke v1beta1/2.

**Comment by krousey** [comment]
@vishh is out today, so I asked @vmarmol if he could look into getting a new heapster image released today.

**Comment by krousey** [comment]
https://github.com/GoogleCloudPlatform/heapster/pull/301 FYI

**Comment by krousey** [comment]
I rebased and added a couple of more commits. One removes swagger specs. I'd like someone to make sure that looks sane. The other updates our use of heapster to version 0.13.0 which was cut to add v1beta3 support.

**Comment by krousey** [comment]
I'll squash once I get the OK.

**Comment by vmarmol** [comment]
I would say to split some of those commits into other PRs, but outside of that the Heapster changes LGTM :)

**Comment by bgrant0607** [comment]
One PR/commit makes it easier to rollback.

**Comment by bgrant0607** [comment]
LGTM. To be merged on Monday morning.
cc @erictune, who will be on call

**Comment by krousey** [comment]
Squashed

**Comment by krousey** [comment]
e2e passed with new heapster image

**Comment by bgrant0607** [comment]
please rebase

**Comment by krousey** [comment]
Done, and why isn't shippable running?

**Comment by bgrant0607** [comment]
Merge conflict would be my guess.

**Comment by krousey** [comment]
The experimental Jenkins builder seems to have failed to fetch this PR to test. Doesn't seem to be a problem with this PR specifically.

**Pull request merged**